Apollo/src/main.cpp

```diff
@@ -0,0 +1,140 @@
+/*                              Autonomous Task                              */
+/*                                                                           */
+/*  This task is used to control your robot during the autonomous phase of   */
+/*  a VEX Competition.                                                       */
+/*                                                                           */
+/*  You must modify the code to add your own robot specific commands here.   */
+/*---------------------------------------------------------------------------*/
+
+void autonomous(void) {
+  
+}
+
+/*---------------------------------------------------------------------------*/
+/*                                                                           */
+/*                              User Control Task                            */
+/*                                                                           */
+/*  This task is used to control your robot during the user control phase of */
+/*  a VEX Competition.                                                       */
+/*                                                                           */
+/*  You must modify the code to add your own robot specific commands here.   */
+/*---------------------------------------------------------------------------*/
+
+void usercontrol(void) {
+  Master.ButtonR1.pressed(toggleFrontMogoLift);
+
+  Master.ButtonUp.pressed(movePlungerOpen);
+  Master.ButtonDown.pressed(movePlungerRest);
+  Master.ButtonLeft.pressed(movePlungerPrep);
+  Master.ButtonRight.pressed(movePlungerScore);
+  Master.ButtonA.pressed(movePlungerPlunge);
+
+  Master.ButtonB.pressed(plungeRing);
+  Master.ButtonX.pressed(togglePlunger);
+
+  //TODO - move to a different function
+  MinesMotorGroup l(leftDriveTop, leftDriveMid, leftDriveBottom);
+  MinesMotorGroup r(rightDriveTop, rightDriveMid, rightDriveBottom);
+  FourWheelDrive d(&l, &r, &Inertial, &Master);
+
+  l.setStopping(brakeType::coast);
+  r.setStopping(brakeType::coast);
+
+  // User control code here, inside the loop
+  while (1) {
+    d.arcadeLoopCall(Master.Axis3.position(), Master.Axis1.position());
+
+    if (Master.ButtonL1.pressing() && backMogoArm.position(rotationUnits::deg) < -50.0)
+    {
+      backMogoArm.spin(directionType::fwd, 100, percentUnits::pct);
+    }
+    else if (Master.ButtonL2.pressing() && backMogoArm.position(rotationUnits::deg) > -500.0)
+    {
+      backMogoArm.spin(directionType::rev, 100, percentUnits::pct);
+    }
+    else 
+    {
+      backMogoArm.stop();
+    }
+
+    wait(20, msec); // Sleep the task for a short amount of time to
+                    // prevent wasted resources.
+  }
+}
+
+
+//
+// Main will set up the competition functions and callbacks.
+//
+int main() {
+
+  // Set up callbacks for autonomous and driver control periods.
+  Competition.autonomous(autonomous);
+  Competition.drivercontrol(usercontrol);
+
+  // Run the pre-autonomous function.
+  pre_auton();
```

Comment:
Would a Comopetition.pre_autonomous() with pre_autom() as a callback for this function work here?